apps/dash-lunar-lander/app.py

```diff
@@ -0,0 +1,438 @@
+        # Adjust the Spacecraft Parameters
+        dcc.Slider(
+            id="m0Slider",
+            min=5000,
+            max=15000,
+            value=10000,
+            step=10,
+            marks={5000: {"label": "Inital Mass"}},
+        ),
+        dcc.Slider(
+            id="c1Slider",
+            min=35000,
+            max=44000 * 1.1,
+            value=44000,
+            step=10,
+            marks={35000: {"label": "Max Thrust"}},
+        ),
+        dcc.Slider(
+            id="isp", min=300, max=330, value=311, step=1, marks={300: {"label": "ISP"}}
+        ),
+        dcc.Slider(
+            id="c3Slider",
+            min=0.01,
+            max=0.1,
+            value=0.0698,
+            step=0.0001,
+            marks={0.01: {"label": "Max Angular Velocity"}},
+        ),
+        dcc.Slider(
+            id="gSlider",
+            min=0.5,
+            max=2,
+            value=1.6229,
+            step=0.0001,
+            marks={0.5: {"label": "Gravity"}},
+        ),
+        html.Div(
+            [
+                html.H5(
+                    "Parameters", style={"width": "25%", "display": "inline-block"}
+                ),
+                html.H5("Outputs", style={"width": "25%", "display": "inline-block"}),
+                html.H5("Adjust Inital Position", style={"display": "inline-block"}),
+            ]
+        ),
+        html.Div(
+            [
+                # Display Current Spacecraft Parameters
+                html.Div(
+                    [
+                        html.P("Inital Mass (kg):", id="m0Out"),
+                        html.P("Max Thrust (N):", id="maThrustOut"),
+                        html.P("ISP (s):", id="ispOut"),
+                        html.P("C3 (rad/s):", id="c3Out"),
+                        html.P("Gravity (N):", id="gravOut"),
+                    ],
+                    style={"width": "24%", "display": "inline-block"},
+                ),
+                html.Div(
+                    [
+                        # Choose Between Different Cost Functions
+                        dcc.RadioItems(
+                            options=[
+                                {"label": "Mass Optimal Control", "value": 1},
+                                {"label": "Time Optimal Control", "value": 2},
+                            ],
+                            id="costFun",
+                            value=1,
+                        ),
+                        # Display Final Cost Functions
+                        html.P("Final  Mass (kg):", id="mfOut"),
+                        html.P("TOF (s):", id="tof"),
+                    ],
+                    style={"width": "24%", "display": "inline-block"},
+                ),
+                # DPad for Adjusting the Spacecrafts Initial Position
+                html.Div(
+                    [
+                        html.Button(
+                            "Left", id="Left", style={"width": "100%", "height": "100%"}
```

Comment:
I'm okay to leave minimal inline-css (i.e. `style={}`) for specific component but for code cleaness it's always good habit to add your css in style.css, especially when you have multiple css which could be pooled together

something like 
```
#Up, #Down, #Left {
width: 100%, height: 100%;
}

```

Reply:
good call, will update with next commit